File: backend/src/main/java/com/quizmaster/model/Flashcard.java
package com.quizmaster.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "flashcards")
public class Flashcard {
    
    @Id
    private String id;
    
    @Indexed
    private String title;
    
    private String description;
    
    private List<String> tags;
    
    private String createdBy;
    
    private boolean isPublic;
    
    private List<Card> cards;
    
    @CreatedDate
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    private LocalDateTime updatedAt;
    
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Card {
        private String id;
        private String front;
        private String back;
        private String imageUrl;
        private int position;
    }
}

File: frontend/src/types/flashcard.ts
export interface FlashcardCard {
  id: string
  front: string
  back: string
  imageUrl?: string
  position: number
}

export interface Flashcard {
  id: string
  title: string
  description?: string
  createdBy: string
  createdAt: string
  updatedAt?: string
  isPublic: boolean
  tags: string[]
  cards: FlashcardCard[]
}

export interface FlashcardStudySubmission {
  cardResults: Record<string, boolean> // cardId -> remembered (true/false)
  timeSpent: number // in seconds
}

export interface FlashcardStudy {
  id: string
  flashcardId: string
  flashcardTitle: string
  totalCards: number
  cardsStudied: number
  cardsRemembered: number
  cardsToReview: number
  completed: boolean
  timeSpent: number // in seconds
  startedAt: string
  completedAt?: string
}
